lib/Dialect/TritonGPU/IR/Dialect.cpp

```diff
@@ -83,6 +83,15 @@ unsigned getTotalElemsPerThread(Type type) {
   if (type.isIntOrIndexOrFloat() || isa<triton::PointerType>(type))
     return 1;
   auto tensorType = cast<RankedTensorType>(type);
+
+  std::optional<LinearLayout> ll = triton::gpu::toLinearLayout(
```

Comment:
All the layout in the Intel GPU dialect have support the LL already.
Try to get the meta data from LL first. Fallback to the legacy layout interface If it doesn't support LL